metrics/grafana/tikv_trouble_shooting.json

```diff
@@ -7222,7 +7222,7 @@
           "steppedLine": false,
           "targets": [
             {
-              "expr": "tikv_engine_pending_compaction_bytes{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", instance=~\"$instance\", db=\"$db\"}",
+              "expr": "10 * sum(rate(tikv_engine_pending_compaction_bytes_sum{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", instance=~\"$instance\", db=\"$db\"}[1m]))",
```

Comment:
Isn't it a guage? Why sum up with an interval?

Reply:
It's a histogram now.

Reply:
Why the change? If it's reported twice within 1m or lose some of the metrics within 1m, the metrics will become either too large or too small? Is it even a compatible change?

Reply:
Because it's the easiest way to report the statistics with some distribution data. Yes the actual reporting will be affected by the reporting frequency (e.g. 10s per reporting so it's needed to scale the counter by 10).

Reply:
But it's inaccurate. The correct way to do this is collect all the metrics in a temporary struct and publish them at once. So no definitions need to be adjusted.

```
let mut stats = CfStats::default();
flush_engine_properties(engine, name, &mut stats);
publish(engine, stats);

fn flush_engine_properties(engine, name, stats) {
    for (i, cf) in engine.cf_names().enumerate() {
        if let Some(v) = engine.get_property(cf, prop_name) {
            stats[i] += v;
        }
    }
}

fn publish(engine, name, stats) {
    for (i, cf) in engine.cf_names().enumerate() {
        REAL_METRICS.with_label_value(&name, cf).set(stats[i])
    }
}
```